### ISSUE TITLE
AzureAD authentication

### ISSUE THREAD

**Issue by ikcalB** [WeblateOrg/weblate#7749]
### Describe the issue

TL;DR: AzureAD authentication does not work.

I suspect either the redirection is wrongly configured, or maybe a bug
When trying to login, the redirection to `login.microsoft.com` seems to work. The first time I was asked to allow the `User.Read` permission.

When I try to sign in on the weblaute UI, the browser keeps reloading constantly. For each reload, I can see a successful authentication in the azure logs.

`docker logs weblate_weblate_1:`
```
nginx stdout | 195.34.137.202 - - [07/Jun/2022:07:51:56 +0000] "GET /accounts/login/?next=/ HTTP/1.1" 200 3840 "https://translate.site.com/" "Mozilla/5.0 (X11; Linux x86_64; rv:100.0) Gecko/20100101 Firefox/100.0"
nginx stdout | 195.34.137.202 - - [07/Jun/2022:07:51:57 +0000] "POST /accounts/login/azuread-oauth2/ HTTP/1.1" 302 5 "https://translate.site.com/" "Mozilla/5.0 (X11; Linux x86_64; rv:100.0) Gecko/20100101 Firefox/100.0"
gunicorn stderr | [2022-06-07 07:51:57,740: WARNING/374] Handled exception: AuthForbidden: Your credentials aren't allowed
nginx stdout | 195.34.137.202 - - [07/Jun/2022:07:51:57 +0000] "GET /accounts/complete/azuread-oauth2/?code=<code...>&state=<state...>&session_state=<session_state ...> HTTP/1.1" 302 5 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:100.0) Gecko/20100101 Firefox/100.0"

nginx stdout | 195.34.137.202 - - [07/Jun/2022:07:51:57 +0000] "GET /accounts/login/ HTTP/1.1" 200 3880 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:100.0) Gecko/20100101 Firefox/100.0"
nginx stdout | 195.34.137.202 - - [07/Jun/2022:07:51:58 +0000] "POST /accounts/login/azuread-oauth2/ HTTP/1.1" 302 5 "https://translate.site.com/" "Mozilla/5.0 (X11; Linux x86_64; rv:100.0) Gecko/20100101 Firefox/100.0"
gunicorn stderr | [2022-06-07 07:51:58,570: WARNING/373] Handled exception: AuthForbidden: Your credentials aren't allowed
```

- I have searched everywhere I could think of, for variations of the `AuthForbidden` type error (github weblate and python-social-auth, stackoverflow, ...)
- looked at the source code
- Issue may be similar to #2497, although I don't think so:
  - access logs on azure show successfull authentications.
  - this issue mentions quoting with double quotes. I've tried with and without single quotes
  - I don't get 401s

### I already tried

- [X] I've read and searched [the documentation](https://docs.weblate.org/).
- [X] I've searched for similar issues in this repository.

### Steps to reproduce the behavior

1. add `WEBLATE_SOCIAL_AUTH_AZUREAD_OAUTH2_KEY` and `WEBLATE_SOCIAL_AUTH_AZUREAD_OAUTH2_SECRET` to the docker-compose.override.yml
2. deploy via docker
3. click sign-in
4. if not already signed into an microsoft online account (with the browser. presumably from other applications using sso) sign in with your credentials

=> browser will reload constantly

### Expected behavior

complete login

### Screenshots

![screen-weblate_does_not_allow](https://user-images.githubusercontent.com/3233280/172331455-44ddcc06-0640-4556-a27a-21cfa45d2196.png)

### How do you run Weblate?

Docker container

### Weblate versions

weblate@590dfdd4751a:/$ weblate list_versions
 * Weblate: 4.12.2
 * Django: 4.0.4
 * siphashc: 2.1
 * translate-toolkit: 3.6.2
 * lxml: 4.6.5
 * Pillow: 9.1.0
 * bleach: 5.0.0
 * python-dateutil: 2.8.2
 * social-auth-core: 4.2.0
 * social-auth-app-django: 5.0.0
 * django-crispy-forms: 1.14.0
 * oauthlib: 3.2.0
 * django-compressor: 4.0
 * djangorestframework: 3.13.1
 * django-filter: 21.1
 * django-appconf: 1.0.5
 * user-agents: 2.2.0
 * filelock: 3.6.0
 * jellyfish: 0.9.0
 * openpyxl: 3.0.9
 * celery: 5.2.6
 * kombu: 5.2.4
 * translation-finder: 2.12
 * weblate-language-data: 2022.2
 * html2text: 2020.1.16
 * pycairo: 1.21.0
 * pygobject: 3.42.1
 * diff-match-patch: 20200713
 * requests: 2.27.1
 * django-redis: 5.2.0
 * hiredis: 2.0.0
 * sentry_sdk: 1.5.12
 * Cython: 0.29.29
 * misaka: 2.1.1
 * GitPython: 3.1.27
 * borgbackup: 1.2.0
 * pyparsing: 3.0.9
 * pyahocorasick: 1.4.4
 * python-redis-lock: 3.7.0
 * charset-normalizer: 2.0.12
 * Python: 3.10.4
 * Git: 2.30.2
 * psycopg2: 2.9.3
 * psycopg2-binary: 2.9.3
 * phply: 1.2.5
 * ruamel.yaml: 0.17.21
 * tesserocr: 2.5.2
 * akismet: 1.1
 * boto3: 1.22.13
 * zeep: 4.1.0
 * aeidon: 1.11
 * iniparse: 0.5
 * mysqlclient: 2.1.0
 * Mercurial: 6.1.2
 * git-svn: 2.30.2
 * git-review: 2.3.1
 * Redis server: 6.2.7
 * PostgreSQL server: 14.3
 * Database backends: django.db.backends.postgresql
 * Cache backends: default:RedisCache, avatar:FileBasedCache
 * Email setup: django.core.mail.backends.smtp.EmailBackend: smtp.office365.com
 * OS encoding: filesystem=utf-8, default=utf-8
 * Celery: redis://cache:6379/1, redis://cache:6379/1, regular
 * Platform: Linux 3.10.0-1160.24.1.el7.x86_64 (x86_64)

### Weblate deploy checks

```shell
weblate@590dfdd4751a:/$ weblate check --deploy
System check identified some issues:

INFOS:
?: (weblate.I021) Error collection is not set up, it is highly recommended for production use
        HINT: https://docs.weblate.org/en/weblate-4.12.2/admin/install.html#collecting-errors
?: (weblate.I028) Backups are not configured, it is highly recommended for production use
        HINT: https://docs.weblate.org/en/weblate-4.12.2/admin/backup.html

System check identified 2 issues (1 silenced).
```

**Comment by ikcalB** [comment]
@nijel looks like that was the case. (when I tested with "wrong" secret, I added a letter. that somehow changed the login result on the azure logs O.o)

why would the azure logs show a successful login with a wrong secret... !? (not user login, but application login)

**Comment by ikcalB** [comment]
that insight is helpful. thanks!
